javascript/ecmascript-2015/promises/introducing-promises.md

```diff
@@ -24,25 +24,17 @@ links:
 ---
 ## Content
 
-A **promise** is an object that represents a value that may or may not be obtained in the future and can be passed around like any other object.
+A **promise** is an object that represents a value that may or may not be obtained in the future and can be passed around like any other value.
 
-You may have come across promises in JavaScript before either in the language itself, libraries such as jQuery (which some would argue are not really promises) and libraries such as q or rsvp.
+### Why use promises?
 
-Some of these libraries conform (and add their own additional functionality as well) to a specification called **Promise/A+**. ES6 bases its promise implementation on this specification.
+It is common in JavaScript to perform tasks that do not complete immediately – for example making an ajax call. Pages would run very slowly if we had to stop everything on a page to wait for an ajax call so JavaScript continues to execute other code.
 
-**Why use promises?**
+However we still need a way to know when our ajax call has completed. One way of approaching this problem is to use callback functions. A callback function is called when the original task has completed.
 
-It is common in JavaScript to perform tasks that do not complete immediately – for example making an ajax call.
+Callbacks work great for simple scenarios such as doing one thing at a time. As soon as we need to do multiple asynchronous tasks it starts getting out of control (e.g. having multiple callbacks and exceptions).
```

Comment:
Maybe:

> The most common approach in JavaScript to perform an async operation is to use a callback with an async mechanism like `fetch` or `setTimeout`. Callbacks work great for simple scenarios such as doing one or two async tasks at a time. As soon as we need to involve more async tasks or increase their complexity with logic and/or error handling, callbacks-based code starts getting difficult to manage.

> With Promises we do not get this issue because promises are inherently reliable (they handle errors for us automatically) and they allow us to write simpler syntax for handling async logic

> With Promises we can write code that pretends like we already have the value, no matter when that value actually arrives. This allows Promises to eliminate time as a concern in our code.

Reply:
This would be a code snippet to demonstrate that but we could show it in a later insight:

### Promises eliminate time as a concern

```js
// Creating a Promise returns immediately
var promise = new Promise(function(resolve, reject) {
 // we call resolve() on success
 // we call reject() on fail
});

// ...
// And we can use that promise anytime and 
// anywhere later to resolve it to a value.
// When creating a promise we are given a
// structure holding our eventual result
// and the promise doesn't care how, or
// even if we use it.
// 
// Getting a value from a promise is fully
// decoupled from what we do with that value
// ...

promise.then(function(value) {
  // here we have the value
}, function(error) { 
  // here we have the error
});
```

### Callbacks do not

```js
// when we perform an async operation
// that involves a callback, we have
// to immediately provide that callback
//
// we cannot decide to handle this 
// operation differently in another place
// or read the async value multiple times
// etc.
getThing(() => {

});
```

Reply:
I've added your suggestions to this insight, but I've changed the format a bit. Let me know if I should make any changes.